pkg/controller/storagecluster/reconcile.go

```diff
@@ -0,0 +1,97 @@
+package storagecluster
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1alpha1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1alpha1"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+)
+
+// Reconcile reads that state of the cluster for a StorageCluster object and makes changes based on the state read
+// and what is in the StorageCluster.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling StorageCluster")
+
+	// Fetch the StorageCluster instance
+	instance := &ocsv1alpha1.StorageCluster{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	// Define a new CephCluster object
+	cephCluster := newCephCluster(instance)
+
+	// Set StorageCluster instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, cephCluster, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if this CephCluster already exists
+	found := &rookCephv1.CephCluster{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cephCluster.Name, Namespace: cephCluster.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new CephCluster", "CephCluster.Namespace", cephCluster.Namespace, "CephCluster	.Name", cephCluster.Name)
+		err = r.client.Create(context.TODO(), cephCluster)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// CephCluster created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// CephCluster already exists - don't requeue
```

Comment:
You don't want to just skip it, you want to run `r.client.Update()` to make sure the desired configuration is always reflected in the existing object.